sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -644,6 +644,8 @@ object PushFoldableIntoBranches extends Rule[LogicalPlan] {
 
   def apply(plan: LogicalPlan): LogicalPlan = plan.transformWithPruning(
     _.containsAnyPattern(CASE_WHEN, IF), ruleId) {
+    case a @ Aggregate(groupingExpressions, _, _)
+      if !groupingExpressions.forall(_.isInstanceOf[NamedExpression]) => a
```

Comment:
This is because the aggregate contains complex grouping expressions after `RedundantAggregates`. So could we fix it there.

Reply:
Thanks. In `PhysicalAggregation`，if the expression is not a NamedExpressions, it add an alias. Also I'm not sure if complex expressions are generated anywhere else other than `RemoveRedundantAggregates`. So is it safer to fix `PushFoldableIntoBranches`, and the impact is smaller?

Reply:
We had a similar issue before: https://issues.apache.org/jira/browse/SPARK-34581

Reply:
Ok, later I will do something like `PullOutGroupingExpressions` after `RemoveRedundantAggregates` triggers Remove.